external/llvm-project/mlir/include/mlir/Dialect/Async/IR/AsyncOps.td

```diff
@@ -27,6 +28,66 @@ include "mlir/Interfaces/SideEffectInterfaces.td"
 class Async_Op<string mnemonic, list<Trait> traits = []> :
     Op<AsyncDialect, mnemonic, traits>;
 
+def Async_LaunchOp :
+  Async_Op<"launch", [DeclareOpInterfaceMethods<CallOpInterface>,
+                      AttrSizedOperandSegments]> {
+  let summary = "Asynchronous launch operation";
+  let description = [{
+    The `async.launch` operation encapsulates a call to a 'kernel' func that
+    may run asynchronously from the caller. A 'async.token' is returned to
+    maintain ordered dependent execution.
+
+    The actual concurrency semantics depends on the dialect lowering to the
+    executable format. Fully sequential execution ("compute0" completes before
+    "compute1" starts) is a completely legal execution.
```

Comment:
This has a semantics.  I think you need to clarify what exactly is the semantics here... i.e. if compute0 and compute1 are 2 launches without an explicit event dependency, then they may execute in any order:  compute0 may start and complete before compute1, compute1 may start and complete before compute0, or they may execute concurrently.